packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -604,3 +604,112 @@ export async function collectFilesForIndex(
     // pick top 100k files below size limit
     return storage.slice(0, Math.min(100000, i))
 }
+
+export async function listFilesWithinDistanceAgainstFile(
```

Comment:
side-note: IIRC, codewhisper (used to?) have similar functionality for its "dependency graph" stuff. I know that "dependency graph" was removed; was its "nearby files" logic also removed?

Reply:
umm, don't think we have them anymore but i will double check this Justin

Reply:
> have similar functionality for its "dependency graph" stuff


and i feel previously code scan use case is to search files with dependency but not file distance

Reply:
but i will double check this, Justin